test/test_roles.py

```diff
@@ -0,0 +1,29 @@
+"""Tests Role class."""
+import pytest
+
+from ansible_compat.roles import Role
+
+
+@pytest.mark.parametrize(
+    ("role_name", "result"),
+    (
+        ("x.y.z", False),  # 3 dots unexpected
+        ("foo-bar", False),  # dash not allowed
+        ("foo_bar", False),  # missing namespace
+        ("ns.foo-bar", False),  # dash not allowed
+        ("ns.foo_bar", True),
+        ("for.foo", False),  # 'for' is python identifier
+        ("ns.for", False),  # 'for' is python identifier
+    ),
+)
+def test_role_fail(role_name: str, result: bool) -> None:
+    """Check if Role.validate returns expected result."""
+    role = Role(role_name)
+    assert role.validate() is result
+
+    # asserts that exception is raised when check=True
+    if not result:
+        with pytest.raises(RuntimeError):
+            Role(role_name, check=True)
+    else:
+        Role(role_name, check=True)
```

Comment:
This should live in a separate test (or be gone for good if we remove the `check` constructor parameter.

Reply:
Here is a challenge: how to avoid duplicating all the 7 tests between two tests? I know that is good to separate tests but it is also good to avoid duplicating data and I do not fancy repeating all the 7 examples for two different tests.